homework4/Routers/Routers/Program.cs

```diff
@@ -0,0 +1,80 @@
+﻿// <copyright file="Program.cs" company="PlaceholderCompany">
+// Copyright (c) PlaceholderCompany. All rights reserved.
+// </copyright>
+
+using Routers;
+
+if (args.Length < 2)
+{
+      System.Console.WriteLine("Enter <input_file> <output_file>");
+      return;
+}
+
+string inputFileName = args[0];
+string outputFileName = args[1];
+
+if (string.IsNullOrWhiteSpace(inputFileName))
+{
+      System.Console.WriteLine("File path cannot be null.");
+      return;
+}
+
+if (!File.Exists(inputFileName))
+{
+      System.Console.WriteLine("File not exists, try again.");
+      return;
+}
+
+if (new FileInfo(inputFileName).Length == 0)
+{
+      System.Console.WriteLine("File is empty, try again.");
+      return;
+}
+
+var network = ReadFromFile(inputFileName);
+if (!network.IsNetworkConnected())
+{
+      System.Console.WriteLine("Graph should be connected!");
+      return;
+}
+
+var newNet = network.PrimAlgorithm();
+WriteToFile(newNet, outputFileName);
+
+Network ReadFromFile(string filename)
+{
+      var network = new Network();
+      using (StreamReader reader = new StreamReader(filename))
```

Comment:
```suggestion
      using var reader = new StreamReader(filename);
```
И дальше без лишних фигурных скобок